### PR TITLE
Add support for STS

### PR DESCRIPTION
**Pull request by andyrzhao**
Added "--uat" boolean flag to perform UAT exchange, converting the
fetched OAuth Access Token to UAT. This flag is designed to be
compatible with oauth and sso, as well as curl and caching.

Added "--user_project" GUAC parameter for use in UAT claim.

Refactored and simplified main.go and tasks.go by introducing
the concept of "taskSettings".

Refactored SSO code path to align with oauth and jwt code path,
so it can take advantage of caching and uat support.

Core UAT exchange logic is in util/uat.go

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -285,6 +285,19 @@ Note: The custom SSO CLI should have the following interface:
 $ /usr/bin/sso me@example.com scope1 scope2
 ```
 
+### --uat
```
Comment (shinfan) [inline README.md:288]: Rename uat with STS.

File: README.md [hunk 2]
```diff
@@ -285,6 +285,19 @@ Note: The custom SSO CLI should have the following interface:
...
+
+If true, exchanges the fetched OAuth access token with an UAT token using
```
Comment (shinfan) [inline README.md:290]: Not necessary oauth access token, can be self-signed access token as well.

File: main.go [hunk 3]
```diff
@@ -261,15 +251,26 @@ func main() {
 		credentials := getCredentialsWithFallback(commonOpts)
 		scope := commonOpts.Scope
 		audience := commonOpts.Audience
+		userProject := commonOpts.UserProject
```
Comment (shinfan) [inline main.go:254]: Can we allow this option when option uat is set?

File: README.md [hunk 4]
```diff
@@ -299,12 +312,20 @@ $ oauth2l fetch --scope cloud-platform,pubsub
...
 
 The single audience to include in the signed JWT token. Required for jwt
-authentication type.
+authentication type. Can also be used for UAT.
 
 ```bash
 $ oauth2l fetch --type jwt --audience https://pubsub.googleapis.com/
 ```
 
+### --user_project
```
Comment (shinfan) [inline README.md:321]: quota_project to be consistent with gcloud

File: README.md (hunk 1 above)
Comment (andyrzhao) [reply README.md:288]: Done

File: README.md (hunk 2 above)
Comment (andyrzhao) [reply README.md:290]: Removed "OAuth" - however, note that STS doesn't support self-signed JWT as of today.

File: README.md (hunk 4 above)
Comment (andyrzhao) [reply README.md:321]: Done

**Pull request merged**